src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -92,6 +103,11 @@ class MetricsBucket(TypedDict):
     value: Any
 
 
+class MetricsBucketBilling(NamedTuple):
+    billing_future: Future[Message[KafkaPayload]]
+    metrics_msg: Message[KafkaPayload]
```

Comment:
nit: Since all we need is an offset and a partition, we could store a tuple `(next_offset, partition)` here instead.

Reply:
We also use `next_offset` in `mark_commit_ready`. I can update the code to make a tuple of 3 items, but I think the current implementation is easier to read (the concept of a message is already in your brain, and you don't have to think about anything else). Another reason to use a tuple may be the memory footprint, I may be wrong but I don't think this is an issue for now.

Reply:
Sounds good to me. Let's add a gauge metric somewhere to keep track of the deque size. If it higher than what we expect we can still revisit this.

Reply:
Good idea, added the metric in [4a03607](https://github.com/getsentry/sentry/pull/40030/commits/4a03607838cdc04f53a2cde93e4b89c5c2bb7ddd). It'd probably be more interesting to get the metric on `submit` operations, but considering how frequent those calls are and the resource consumption of the metrics client (plus the potential rate-limits), I decided to add it in `_bulk_commit`.